### DISCUSSION TITLE
Take the environments into account when calling: `php artisan schedule list`

### DISCUSSION THREAD

**Discussion by Mirnaxvb** [laravel/framework#49143] [Ideas]
### Current behaviour

When you schedule a command with an environment filter on it like so:

```php
$schedule->command('horizon:snapshot')
    ->everyFiveMinutes()
    ->environments(['staging']);
```

and you call `php artisan schedule:list`

It will still output as schedule even though the env is set to `APP_ENV=local`:

```shell
  */5 *  * * *  php artisan horizon:snapshot ................................ Next Due: 4 minutes from now
```

### Solutions

We could solve this in two ways:

#### 1. Filter out based on the environments
We could simply filter on the environments changing this: https://github.com/laravel/framework/blob/10.x/src/Illuminate/Console/Scheduling/ScheduleListCommand.php#L53

```php
$events = collect($schedule->events());
```
to

```php
use Illuminate\Support\Facades\App;

//...

$events = collect($schedule->events())
            ->filter(fn (Event $event) => $event->runsInEnvironment(App::environment()));
```

#### 2. Add the environments to the output

If we don't want this behaviour it would be nice if we at least print the environments / wether its active or not.

#### 3. Other..?

**Comment by ghost** [comment]
3rd: call list with extra flag for current environment?

**Reply by ruedigerm-playster** [reply]
I think 1) is most logical.... if the command is not active under current envivonment, it doesn't need to be listed under `schedule:list`

**Reply by ghost** [reply]
But imagine that you are on local and want to see all routes...and only local ones are displayed... So 3 makes more sense.

**Reply by Mirnaxvb** [reply]
Wouldn't you then say 1) by default show only current environment and a "--all" option so show all commands? Still would opt to print environments in the output then.
